server/src/main/java/io/whitefox/api/deltasharing/loader/DeltaShareTableLoader.java

```diff
@@ -0,0 +1,24 @@
+package io.whitefox.api.deltasharing.loader;
+
+import io.whitefox.persistence.memory.PTable;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+@ApplicationScoped
+public class DeltaShareTableLoader implements TableLoader<DeltaSharedTable> {
+
+  private final ConcurrentMap<String, DeltaSharedTable> cache;
+
+  public DeltaShareTableLoader() {
+    this.cache = new ConcurrentHashMap<>();
+  }
+
+  @Override
+  public CompletionStage<DeltaSharedTable> loadTable(PTable pTable) {
```

Comment:
Why do you do that async? This operation is synchronous, you’re dealing with asynchronicity inside DeltaSharedTable